### PR TITLE
feat: add rstcheck

### PR DESCRIPTION
**Pull request by polyzen**
For some reason the error messages for code blocks have the `($language)` truncated.

Output at command-line:
`bad_code.rst:9: (ERROR/3) (python) unexpected EOF while parsing`

Output from `:LspDocumentDiagnostics`:
`bad_code.rst|9 col 1| Error:unexpected EOF while parsing`

### CODE REVIEW

**Comment by tsuyoshicho** [comment]
Thanks, I check it.

File: config/efm-langserver/config.yaml [hunk 1]
```diff
@@ -235,6 +235,15 @@ tools:
...
 
+  rstcheck-lint: &rstcheck-lint
+    lint-command: 'rstcheck -'
+    lint-stdin: true
+    lint-formats:
+      - '%f:%l: (%tNFO/%r) %m'
+      - '%f:%l: (%tARNING/%r) %m'
+      - '%f:%l: (%tRROR/%r) %m'
+      - '%f:%l: (%tEVERE/%r) %m'
+
```
Comment (tsuyoshicho) [inline config/efm-langserver/config.yaml:246]:
```suggestion
    root-markers:
      - .rstcheck.cfg

```

add root marker

File: README.md [hunk 2]
```diff
@@ -59,8 +59,9 @@ This plugin setting up efm-langserver to work.
...
 | [restructuredtext-lint](https://pypi.org/project/restructuredtext-lint/) | rst                                                                          | lint               | pip                  | not need config                          |                                                                                                                                                                                                                                                                                                                         |
+| [rstcheck](https://github.com/myint/rstcheck)                            | rst                                                                          | lint               | pip                  | not need config                          |                                                                                                                                                                                                                                                                                                                         |
```
Comment (tsuyoshicho) [inline README.md:62]:
```suggestion
| [rstcheck](https://github.com/myint/rstcheck)                            | rst                                                                          | lint               | pip                  | not need config / user config affected   | `.rstcheck.cfg` in the project root.                                         |
```

note for config.
(and cell width fix...)

**Review by tsuyoshicho — CHANGES_REQUESTED** [review]
Add config info.

File: config/efm-langserver/config.yaml [hunk 3]
```diff
@@ -235,6 +235,15 @@ tools:
...
       - '%tRROR %f:%l %m'
 
+  rstcheck-lint: &rstcheck-lint
+    lint-command: 'rstcheck -'
+    lint-stdin: true
+    lint-formats:
+      - '%f:%l: (%tNFO/%r) %m'
+      - '%f:%l: (%tARNING/%r) %m'
+      - '%f:%l: (%tRROR/%r) %m'
+      - '%f:%l: (%tEVERE/%r) %m'
```
Comment (tsuyoshicho) [inline config/efm-langserver/config.yaml:245]:
```suggestion
      - '%f:%l: (%tNFO/%*[A-Z0-9]) %m'
      - '%f:%l: (%tARNING/%*[A-Z0-9]) %m'
      - '%f:%l: (%tRROR/%*[A-Z0-9]) %m'
      - '%f:%l: (%tEVERE/%*[A-Z0-9]) %m'
```

```
	%r		matches the "rest" of a single-line file message %O/P/Q
	%*{conv}	any scanf non-assignable conversion
```

message `<file>:<line>: (ERROR/3) (python) foobar...` and `(%tRROR/%r)` matcher, match `(ERROR/3) (python)` as many as possible matching result.

**Review by tsuyoshicho — CHANGES_REQUESTED** [review]

**Comment by tsuyoshicho** [comment]
Thanks for your PR.

I fixed and updated it here and merged it.

**Comment by polyzen** [comment]
Thank you!

**Pull request merged**